### ISSUE TITLE
pgdriver: unexplicit error

### ISSUE THREAD

**Issue by zekth** [uptrace/bun#560]
I faced a really annoying issue lately with the driver. Integrating with TLS in a k8s environment the connection kept telling me `bad connection`. By debugging locally using some infra tricks to hit the remote cluster i ended up having more detailled errors using `dlv`.
Issue is here: https://github.com/uptrace/bun/blob/a327b2ae216abb55a705626296c0cdbf8d648697/driver/pgdriver/driver.go#L165
It does not propagate the error. One proposal would be like the option:

`bundebug.WithVerbose(true)`

A way to print verbose error in the std-out for easier debugging on remote environment. wdyt?

**Comment by vmihailenco** [comment]
What was the underlying/masked error? Usually zero-length writes mean  that the connection is broken...

**Comment by zekth** [comment]
I had issues with invalid certs, invalid DSN and after i had to add manually set the Servername. like:
```golang
		t := &tls.Config{InsecureSkipVerify: false, MinVersion: tls.VersionTLS12}
		certs := x509.NewCertPool()
		pemData, err := ioutil.ReadFile(cfg.Database.CABundlePath)
		if err != nil {
			logger.Sugar().Errorf("cannot read CA bundle: %v", err)
			os.Exit(-1)
		}
		certs.AppendCertsFromPEM(pemData)
		t.RootCAs = certs
		t.ServerName = cfg.Database.Hostname
		sqldb = sqltrace.OpenDB(pgdriver.NewConnector(
			pgdriver.WithDSN(cfg.Database.DSN()),
			pgdriver.WithTLSConfig(t),
		))
```

**Comment by vmihailenco** [comment]
I think logging the error should be enough in most cases. Creating an option is probably not worth it.

**Comment by zekth** [comment]
Closing as fixed in #564
